ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -163,7 +163,10 @@ PullQueryResult executePullQuery(
           queryAnalyzer.analyze(statement.getStatement(), Optional.empty()),
           new PullQueryExecutionUtil.ColumnReferenceRewriter()::process
       );
-      final KsqlConfig ksqlConfig = sessionConfig.getConfig(true);
+      // Do not set sessionConfig.getConfig to true! The copying is inefficient and slows down pull
+      // query performance significantly.  Instead use PullPlannerOptions which check overrides
+      // deliberately.
+      final KsqlConfig ksqlConfig = sessionConfig.getConfig(false);
```

Comment:
yikes. i wonder if there's a way for us to return a "view" when we call `getConfig(true)` instead of copying. This has bitten us twice 😢

Reply:
Yeah, I think rather than copy, we should more or less do what we do in this PR.  Don't copy, but just check the overrides first.